internal/common/common.go

```diff
@@ -137,3 +138,19 @@ func WriteFileAtomic(filename string, perm os.FileMode, data []byte) error {
 			return err
 		})
 }
+
+var (
+	once     sync.Once
+	hostname string
+)
+
+func GetHostname() string {
+	once.Do(func() {
+		var err error
+		hostname, err = os.Hostname()
+		if err != nil {
+			hostname = ""
+		}
+	})
+	return hostname
+}
```

Comment:
I'll put this in a dedicated global struct instead of having two global variables with very generic (`once`) names.